### PR TITLE
Use GitHub Actions instead of Travis CI

### PR DESCRIPTION
**Pull request by josephfrazier**
Travis CI is no longer providing CI minutes for open source projects: https://news.ycombinator.com/item?id=25338983

It's also been pretty slow: the build for https://github.com/slevithan/xregexp/pull/308
hasn't started in 13 minutes

The config file is based on the template at https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-nodejs#specifying-the-nodejs-version

### CODE REVIEW

**Comment by slevithan** [comment]
Can we also replace the Travis CI build status badge at the top of README.md with something equivalent?

**Comment by josephfrazier** [comment]
Good call, I'll try to see if I can find something.

It looks like we'll need to manually enable the workflow once this is merged: https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/disabling-and-enabling-a-workflow

But, I'm still working on fixing a bug in it: https://github.com/josephfrazier/xregexp/actions/runs/407379254

**Comment by slevithan** [comment]
> Good call, I'll try to see if I can find something.

If GitHub indicates build status in some other way on the repo when using GitHub Actions, then we can just kill the badge in `README.md` (which would be for the best since the readme shows up on npm and elsewhere).

**Comment by slevithan** [comment]
> Good call, I'll try to see if I can find something.

Maybe it's for the best to just kill the badge anyway, since the readme shows up on npm and elsewhere.

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # XRegExp 4.4.0
 
-[![Build Status](https://travis-ci.org/slevithan/xregexp.svg?branch=master)](https://travis-ci.org/slevithan/xregexp)
+[![Build Status](https://github.com/slevithan/xregexp/workflows/.github/workflows/build.yml/badge.svg)](https://github.com/slevithan/xregexp/actions)
```
Comment (slevithan) [inline README.md:3]: nice

**Review by slevithan — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (josephfrazier) [reply README.md:3]: Thanks, I _think_ it should work, but we can check once it's merged.

**Comment by josephfrazier** [comment]
> But, I'm still working on fixing a bug in it: [josephfrazier/xregexp/actions/runs/407379254](https://github.com/josephfrazier/xregexp/actions/runs/407379254)

ok, looks like we're good now: https://github.com/josephfrazier/xregexp/runs/1515139490?check_suite_focus=true

File: README.md (hunk 1 above)
Comment (josephfrazier) [reply README.md:3]: fixed in https://github.com/slevithan/xregexp/pull/310

**Pull request merged**